lib/wraplibsnarkgadgets.cpp

```diff
@@ -0,0 +1,139 @@
+//https://gist.github.com/kobigurk/24c25e68219df87c348f1a78db51bb52
+
+#include <iostream>
+
+//#include "wraplibsnarkgadgets.cpp"
+
+/*#include "libsnark/gadgetlib1/gadget.hpp"
+#include "libsnark/gadgetlib1/protoboard.hpp"
+#include "libff/common/default_types/ec_pp.hpp"
+#include "libsnark/reductions/r1cs_to_qap/r1cs_to_qap.hpp"
+
+#include <libsnark/common/data_structures/merkle_tree.hpp>
+#include <libsnark/gadgetlib1/gadgets/basic_gadgets.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/hash_io.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/sha256/sha256_components.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/sha256/sha256_gadget.hpp>*/
+#include "./sha256_ethereum.cpp"
+
+//r1cs to json
+#include <libsnark/zk_proof_systems/ppzksnark/r1cs_ppzksnark/r1cs_ppzksnark.hpp>
+
+
+using namespace libsnark;
+using namespace libff;
+
+using std::vector;
+
+
+template<typename FieldT>
+std::string r1cs_to_json(protoboard<FieldT> pb, uint input_variables)
+    {  
+    // output inputs, right now need to compile with debug flag so that the `variable_annotations`
+    // exists. Having trouble setting that up so will leave for now.
+    r1cs_constraint_system<FieldT> constraints = pb.get_constraint_system();
+    std::stringstream ss;
+    ss << "\n{\"variables\":[";
+
+    for (size_t i = 0; i < input_variables + 1; ++i)
+    {   
+        ss << '"' << constraints.variable_annotations[i].c_str() << '"';
+        if (i < input_variables ) {
+            ss << ", ";
+        }
+    }
+    ss << "],\n";
+    ss << "\"constraints\":[";
+     
+    for (size_t c = 0; c < constraints.num_constraints(); ++c)
+    {   
+        ss << "[";// << "\"A\"=";
+        constraint_to_json(constraints.constraints[c].a, ss);
```

Comment:
This function does not seem to be defined anywhere, did you forget to include it in the PR?